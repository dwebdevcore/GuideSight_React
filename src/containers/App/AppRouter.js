import React, { Component } from 'react';
import { Route } from 'react-router-dom';
import asyncComponent from '../../helpers/AsyncFunc';
import customRoutes from '../../customApp/router';

const routes = [
  {
    path: '',
    component: asyncComponent(() => import('../Widgets/index.js')),
  },
  {
    path: 'inbox',
    component: asyncComponent(() => import('../Mail')),
  },
  {
    path: 'mailbox',
    component: asyncComponent(() => import('../Mail')),
  },
  {
    path: 'calendar',
    component: asyncComponent(() => import('../Calendar/Calendar')),
  },
  {
    path: 'googlemap',
    component: asyncComponent(() => import('../Map/GoogleMap/googleMap')),
  },
  {
    path: 'leafletmap',
    component: asyncComponent(() => import('../Map/Leaflet/leaflet')),
  },
  {
    path: 'table_ant',
    component: asyncComponent(() => import('../Tables/antTables')),
  },
  {
    path: 'allFormComponent',
    component: asyncComponent(() => import('../Forms/allComponents/')),
  },
  {
    path: 'InputField',
    component: asyncComponent(() => import('../Forms/Input')),
  },
  {
    path: 'editor',
    component: asyncComponent(() => import('../Forms/editor/')),
  },
  {
    path: 'stepperForms',
    component: asyncComponent(() => import('../Forms/StepperForms/')),
  },
  {
    path: 'FormsWithValidation',
    component: asyncComponent(() => import('../Forms/FormsWithValidation')),
  },
  {
    path: 'progress',
    component: asyncComponent(() => import('../Forms/Progress')),
  },
  {
    path: 'ladda_button',
    component: asyncComponent(() => import('../Forms/LaddaButton')),
  },
  {
    path: 'button',
    component: asyncComponent(() => import('../Forms/Button')),
  },
  {
    path: 'tab',
    component: asyncComponent(() => import('../Forms/Tab')),
  },
  {
    path: 'autocomplete',
    component: asyncComponent(() => import('../Forms/AutoComplete')),
  },
  {
    path: 'checkbox',
    component: asyncComponent(() => import('../Forms/Checkbox')),
  },
  {
    path: 'radiobox',
    component: asyncComponent(() => import('../Forms/Radiobox/')),
  },
  {
    path: 'selectbox',
    component: asyncComponent(() => import('../Forms/Select/')),
  },
  {
    path: 'transfer',
    component: asyncComponent(() => import('../Forms/Transfer/')),
  },
  {
    path: 'gridLayout',
    component: asyncComponent(() => import('../Box/GridLayout')),
  },
  {
    path: 'notes',
    component: asyncComponent(() => import('../Notes')),
  },
  {
    path: 'todo',
    component: asyncComponent(() => import('../Todo')),
  },
  {
    path: 'articles',
    component: asyncComponent(() => import('../FirestoreCRUD/Article')),
  },
  {
    path: 'investors',
    component: asyncComponent(() => import('../FirestoreCRUD/Investor')),
  },
  {
    path: 'contacts',
    component: asyncComponent(() => import('../Contacts')),
  },
  {
    path: 'alert',
    component: asyncComponent(() => import('../Feedback/Alert')),
  },
  {
    path: 'modal',
    component: asyncComponent(() => import('../Feedback/Modal/')),
  },
  {
    path: 'message',
    component: asyncComponent(() => import('../Feedback/Message')),
  },
  {
    path: 'notification',
    component: asyncComponent(() => import('../Feedback/Notification')),
  },
  {
    path: 'Popconfirm',
    component: asyncComponent(() => import('../Feedback/Popconfirm')),
  },
  {
    path: 'spin',
    component: asyncComponent(() => import('../Feedback/Spin')),
  },
  {
    path: 'shuffle',
    component: asyncComponent(() => import('../Shuffle')),
  },
  {
    path: 'affix',
    component: asyncComponent(() => import('../Navigation/affix')),
  },
  {
    path: 'breadcrumb',
    component: asyncComponent(() =>
      import('../Uielements/Breadcrumb/breadcrumb')
    ),
  },
  {
    path: 'backToTop',
    component: asyncComponent(() => import('../Navigation/backToTop')),
  },
  {
    path: 'dropdown',
    component: asyncComponent(() => import('../Uielements/Dropdown/dropdown')),
  },
  {
    path: 'op_badge',
    component: asyncComponent(() => import('../Uielements/Badge')),
  },
  {
    path: 'op_card',
    component: asyncComponent(() => import('../Uielements/Card')),
  },
  {
    path: 'op_carousel',
    component: asyncComponent(() => import('../Uielements/Carousel')),
  },
  {
    path: 'op_collapse',
    component: asyncComponent(() => import('../Uielements/Collapse')),
  },
  {
    path: 'op_tooltip',
    component: asyncComponent(() => import('../Uielements/Tooltip/')),
  },
  {
    path: 'rating',
    component: asyncComponent(() => import('../Uielements/rating/')),
  },
  {
    path: 'tree',
    component: asyncComponent(() => import('../Uielements/Tree/')),
  },
  {
    path: 'op_tag',
    component: asyncComponent(() => import('../Uielements/Tag')),
  },
  {
    path: 'op_timeline',
    component: asyncComponent(() => import('../Uielements/Timeline')),
  },
  {
    path: 'op_popover',
    component: asyncComponent(() => import('../Uielements/Popover')),
  },
  {
    path: 'googleChart',
    component: asyncComponent(() => import('../Charts/googleChart')),
  },
  {
    path: 'reecharts',
    component: asyncComponent(() => import('../Charts/recharts')),
  },
  {
    path: 'reactVis',
    component: asyncComponent(() => import('../Charts/reactVis')),
  },
  {
    path: 'menu',
    component: asyncComponent(() => import('../Navigation/menu')),
  },
  {
    path: 'ReactChart2',
    component: asyncComponent(() => import('../Charts/reactChart2')),
  },
  {
    path: 'pagination',
    component: asyncComponent(() =>
      import('../Uielements/Pagination/pagination')
    ),
  },
  {
    path: 'reactTrend',
    component: asyncComponent(() => import('../Charts/reactTrend')),
  },
  {
    path: 'card',
    component: asyncComponent(() => import('../Ecommerce/card')),
  },
  {
    path: 'cart',
    component: asyncComponent(() => import('../Ecommerce/cart')),
  },
  {
    path: 'checkout',
    component: asyncComponent(() => import('../Ecommerce/checkout')),
  },
  {
    path: 'shop',
    component: asyncComponent(() =>
      import('../Ecommerce/algolia/instantSearch')
    ),
  },
  {
    path: 'reactDates',
    component: asyncComponent(() =>
      import('../AdvancedUI/ReactDates/reactDates')
    ),
  },
  {
    path: 'codeMirror',
    component: asyncComponent(() => import('../AdvancedUI/codeMirror')),
  },
  {
    path: 'uppy',
    component: asyncComponent(() => import('../AdvancedUI/uppy')),
  },
  {
    path: 'dropzone',
    component: asyncComponent(() => import('../AdvancedUI/dropzone')),
  },
  {
    path: 'youtubeSearch',
    component: asyncComponent(() => import('../YoutubeSearch')),
  },
  {
    path: 'frappeChart',
    component: asyncComponent(() => import('../Charts/frappeChart')),
  },
  {
    path: 'invoice/:invoiceId',
    component: asyncComponent(() => import('../Invoice/singleInvoice')),
  },
  {
    path: 'invoice',
    component: asyncComponent(() => import('../Invoice')),
  },
  {
    path: 'chat',
    component: asyncComponent(() => import('../Chat')),
  },
  ...customRoutes,
];

class AppRouter extends Component {
  render() {
    const { url, style } = this.props;
    return (
      <div style={style}>
        {routes.map(singleRoute => {
          const { path, exact, ...otherProps } = singleRoute;
          return (
            <Route
              exact={exact === false ? false : true}
              key={singleRoute.path}
              path={`${url}/${singleRoute.path}`}
              {...otherProps}
            />
          );
        })}
      </div>
    );
  }
}

export default AppRouter;
