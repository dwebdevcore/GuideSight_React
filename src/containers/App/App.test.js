import AppRouter from './AppRouter';
import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import Adapter from 'enzyme-adapter-react-16';
const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

Enzyme.configure({ adapter: new Adapter() });

describe('<AppRouter />', () => {
  it('renders with default props', () => {
    const store = mockStore();
    const wrapper = shallow(<AppRouter store={store} />);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
