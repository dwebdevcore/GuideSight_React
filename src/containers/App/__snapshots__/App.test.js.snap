// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AppRouter /> renders with default props 1`] = `
<div>
  <Route
    component={[Function]}
    exact={true}
    path="undefined/"
  />
  <Route
    component={[Function]}
    exact={true}
    key="inbox"
    path="undefined/inbox"
  />
  <Route
    component={[Function]}
    exact={true}
    key="mailbox"
    path="undefined/mailbox"
  />
  <Route
    component={[Function]}
    exact={true}
    key="calendar"
    path="undefined/calendar"
  />
  <Route
    component={[Function]}
    exact={true}
    key="googlemap"
    path="undefined/googlemap"
  />
  <Route
    component={[Function]}
    exact={true}
    key="leafletmap"
    path="undefined/leafletmap"
  />
  <Route
    component={[Function]}
    exact={true}
    key="table_ant"
    path="undefined/table_ant"
  />
  <Route
    component={[Function]}
    exact={true}
    key="allFormComponent"
    path="undefined/allFormComponent"
  />
  <Route
    component={[Function]}
    exact={true}
    key="InputField"
    path="undefined/InputField"
  />
  <Route
    component={[Function]}
    exact={true}
    key="editor"
    path="undefined/editor"
  />
  <Route
    component={[Function]}
    exact={true}
    key="stepperForms"
    path="undefined/stepperForms"
  />
  <Route
    component={[Function]}
    exact={true}
    key="FormsWithValidation"
    path="undefined/FormsWithValidation"
  />
  <Route
    component={[Function]}
    exact={true}
    key="progress"
    path="undefined/progress"
  />
  <Route
    component={[Function]}
    exact={true}
    key="ladda_button"
    path="undefined/ladda_button"
  />
  <Route
    component={[Function]}
    exact={true}
    key="button"
    path="undefined/button"
  />
  <Route
    component={[Function]}
    exact={true}
    key="tab"
    path="undefined/tab"
  />
  <Route
    component={[Function]}
    exact={true}
    key="autocomplete"
    path="undefined/autocomplete"
  />
  <Route
    component={[Function]}
    exact={true}
    key="checkbox"
    path="undefined/checkbox"
  />
  <Route
    component={[Function]}
    exact={true}
    key="radiobox"
    path="undefined/radiobox"
  />
  <Route
    component={[Function]}
    exact={true}
    key="selectbox"
    path="undefined/selectbox"
  />
  <Route
    component={[Function]}
    exact={true}
    key="transfer"
    path="undefined/transfer"
  />
  <Route
    component={[Function]}
    exact={true}
    key="gridLayout"
    path="undefined/gridLayout"
  />
  <Route
    component={[Function]}
    exact={true}
    key="notes"
    path="undefined/notes"
  />
  <Route
    component={[Function]}
    exact={true}
    key="todo"
    path="undefined/todo"
  />
  <Route
    component={[Function]}
    exact={true}
    key="contacts"
    path="undefined/contacts"
  />
  <Route
    component={[Function]}
    exact={true}
    key="alert"
    path="undefined/alert"
  />
  <Route
    component={[Function]}
    exact={true}
    key="modal"
    path="undefined/modal"
  />
  <Route
    component={[Function]}
    exact={true}
    key="message"
    path="undefined/message"
  />
  <Route
    component={[Function]}
    exact={true}
    key="notification"
    path="undefined/notification"
  />
  <Route
    component={[Function]}
    exact={true}
    key="Popconfirm"
    path="undefined/Popconfirm"
  />
  <Route
    component={[Function]}
    exact={true}
    key="spin"
    path="undefined/spin"
  />
  <Route
    component={[Function]}
    exact={true}
    key="shuffle"
    path="undefined/shuffle"
  />
  <Route
    component={[Function]}
    exact={true}
    key="affix"
    path="undefined/affix"
  />
  <Route
    component={[Function]}
    exact={true}
    key="breadcrumb"
    path="undefined/breadcrumb"
  />
  <Route
    component={[Function]}
    exact={true}
    key="backToTop"
    path="undefined/backToTop"
  />
  <Route
    component={[Function]}
    exact={true}
    key="dropdown"
    path="undefined/dropdown"
  />
  <Route
    component={[Function]}
    exact={true}
    key="op_badge"
    path="undefined/op_badge"
  />
  <Route
    component={[Function]}
    exact={true}
    key="op_card"
    path="undefined/op_card"
  />
  <Route
    component={[Function]}
    exact={true}
    key="op_carousel"
    path="undefined/op_carousel"
  />
  <Route
    component={[Function]}
    exact={true}
    key="op_collapse"
    path="undefined/op_collapse"
  />
  <Route
    component={[Function]}
    exact={true}
    key="op_tooltip"
    path="undefined/op_tooltip"
  />
  <Route
    component={[Function]}
    exact={true}
    key="rating"
    path="undefined/rating"
  />
  <Route
    component={[Function]}
    exact={true}
    key="tree"
    path="undefined/tree"
  />
  <Route
    component={[Function]}
    exact={true}
    key="op_tag"
    path="undefined/op_tag"
  />
  <Route
    component={[Function]}
    exact={true}
    key="op_timeline"
    path="undefined/op_timeline"
  />
  <Route
    component={[Function]}
    exact={true}
    key="op_popover"
    path="undefined/op_popover"
  />
  <Route
    component={[Function]}
    exact={true}
    key="googleChart"
    path="undefined/googleChart"
  />
  <Route
    component={[Function]}
    exact={true}
    key="reecharts"
    path="undefined/reecharts"
  />
  <Route
    component={[Function]}
    exact={true}
    key="reactVis"
    path="undefined/reactVis"
  />
  <Route
    component={[Function]}
    exact={true}
    key="menu"
    path="undefined/menu"
  />
  <Route
    component={[Function]}
    exact={true}
    key="ReactChart2"
    path="undefined/ReactChart2"
  />
  <Route
    component={[Function]}
    exact={true}
    key="pagination"
    path="undefined/pagination"
  />
  <Route
    component={[Function]}
    exact={true}
    key="reactTrend"
    path="undefined/reactTrend"
  />
  <Route
    component={[Function]}
    exact={true}
    key="card"
    path="undefined/card"
  />
  <Route
    component={[Function]}
    exact={true}
    key="cart"
    path="undefined/cart"
  />
  <Route
    component={[Function]}
    exact={true}
    key="checkout"
    path="undefined/checkout"
  />
  <Route
    component={[Function]}
    exact={true}
    key="shop"
    path="undefined/shop"
  />
  <Route
    component={[Function]}
    exact={true}
    key="reactDates"
    path="undefined/reactDates"
  />
  <Route
    component={[Function]}
    exact={true}
    key="codeMirror"
    path="undefined/codeMirror"
  />
  <Route
    component={[Function]}
    exact={true}
    key="uppy"
    path="undefined/uppy"
  />
  <Route
    component={[Function]}
    exact={true}
    key="dropzone"
    path="undefined/dropzone"
  />
  <Route
    component={[Function]}
    exact={true}
    key="youtubeSearch"
    path="undefined/youtubeSearch"
  />
  <Route
    component={[Function]}
    exact={true}
    key="frappeChart"
    path="undefined/frappeChart"
  />
  <Route
    component={[Function]}
    exact={true}
    key="invoice/:invoiceId"
    path="undefined/invoice/:invoiceId"
  />
  <Route
    component={[Function]}
    exact={true}
    key="invoice"
    path="undefined/invoice"
  />
  <Route
    component={[Function]}
    exact={true}
    key="chat"
    path="undefined/chat"
  />
  <Route
    component={[Function]}
    exact={true}
    key="githubSearch"
    path="undefined/githubSearch"
  />
  <Route
    component={[Function]}
    exact={true}
    key="blank_page"
    path="undefined/blank_page"
  />
</div>
`;
